### PR TITLE
Group transactions by month, and format currency

### LINKED ISSUE #167: On the Projections page, group transactions by month
Currently all transactions are shown in the same table, with no grouping.

### LINKED ISSUE #177: In Projections, format numbers with spaces (1 000 000) and without decimals.
Decimals are not relevant to Projections, and it would be good to have some spacing between the numbers.

### PR DESCRIPTION
**Pull request by FredrikAugust**
Fixes #167.
Fixes #177.

### CODE REVIEW

File: frontend/src/components/organism/Projections.tsx [hunk 1]
```diff
@@ -33,49 +34,88 @@ const Projections: React.FC<{ className?: string }> = ({ className }) => {
...
-          </tr>
+          <React.Fragment key={`${t.id}-${i}`}>
+            {i === 0 ||
+            (arr[i - 1] &&
+              !moment(arr[i - 1].date, moment.ISO_8601).isSame(
+                date,
+                'month'
+              )) ? (
+              <h3>{date.format('MMMM YYYY')}</h3>
+            ) : null}
```
Comment (FredrikAugust) [inline frontend/src/components/organism/Projections.tsx]: If it's the first month, or if the month is not the same as the previous one.

File: frontend/src/components/organism/Projections.tsx [hunk 2]
```diff
@@ -33,49 +34,88 @@ const Projections: React.FC<{ className?: string }> = ({ className }) => {
...
+  .projection-table {
+    margin-top: 2em;
+  }
+
+  .projection-header {
+    strong {
+      font-weight: 600;
+    }
+
+    position: sticky;
```
Comment (FredrikAugust) [inline frontend/src/components/organism/Projections.tsx]: The header will follow you down as you scroll so you don't lose track of what field is what.

**Comment by flyrev** [comment]
Thanks!

Since we are trying to follow the "one PR per issue" principle, and since we are still waiting for input on https://github.com/Kodeworks/liquidator/issues/177 - could you amend your PR so that it does not add any particular formatting?

**Comment by flyrev** [comment]
One other request (sorry) - could you add a screenshot illustrating how it looks in the browser? That way you can get input from everyone, not just those who have the project checked out and running locally. (Believe it or not, I read a lot of pull requests on my phone these days! 😄 )

**Comment by FredrikAugust** [comment]
Regarding your second to last comment; I have only applied the old function for currency formatting used elsewhere in the application, and not "changed" anything in that matter. Would it be fine to keep this in this PR, and rather create a new PR for updating how the formatting function formats currency at a later date, when that discussion is resolved? That way we don't have to update any other code when/if we decide to change that, and rather only change the `currencyFormat` function.

Sadly, I didn't commit the addition of the `currencyFormat`-call in its own commit (I apologize, and I will try to get better at this), so I can't simply cherry-pick it out.

If you still believe it is worth moving it out, let me know, and I will do it.

**Comment by FredrikAugust** [comment]
Screenshots of projections page.

localhost_8000_projections (1)
localhost_8000_projections

**Comment by flyrev** [comment]
That looks awesome. :1st_place_medal: 

Yes, feel free to merge it. :smile:

**Comment by flyrev** [comment]
Wow, that heart reaction was quick. :)

**Pull request merged**